artifactory/commands/transferinstall/installplugin.go

```diff
@@ -0,0 +1,347 @@
+package transferinstall
+
+import (
+	"fmt"
+	downloadutils "github.com/jfrog/build-info-go/utils"
+	"github.com/jfrog/gofrog/version"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/pkg/errors"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+const (
+	minArtifactoryVersion = "2.9.0" // for reload api (version api is from 2.2.2)
+	pluginReloadRestApi   = "api/plugins/reload"
+	jHomeEnvVar           = "JFROG_HOME"
+	latest                = "[RELEASE]"
+)
+
+var (
+	defalutSearchPath = filepath.Join("opt", "jfrog")
+	// Plugin directory locations
+	OriginalDirPath = PluginFileItem{"artifactory", "etc", "plugins"}
+	V7DirPath       = PluginFileItem{"artifactory", "var", "etc", "artifactory", "plugins"}
+	// Error types
+	EmptyUrlErr            = errors.Errorf("Base download URL must be provided to allow file downloads.")
+	NotValidDestinationErr = errors.Errorf("Can't find plugin directory with the provided information, this command must run on the Artifactory server.")
+	minVerErr              = errorutils.CheckErrorf("This operation requires Artifactory version %s or higher", minArtifactoryVersion)
+)
+
+type PluginFileItem []string
+type PluginFiles []PluginFileItem
+
+// Get the name componenet of the item
+func (f *PluginFileItem) Name() string {
+	size := len(*f)
+	if size == 0 {
+		return ""
+	}
+	return (*f)[size-1]
+}
+
+// Get the directory list of the item, ignore empty entries
+func (f *PluginFileItem) Dirs() *PluginFileItem {
+	dirs := PluginFileItem{}
+	for i := 0; i < len(*f)-1; i++ {
+		dir := (*f)[i]
+		if dir != "" {
+			dirs = append(dirs, dir)
+		}
+	}
+	return &dirs
+}
+
+// Split and get the componenets of the item
+func (f *PluginFileItem) SplitNameAndDirs() (string, *PluginFileItem) {
+	return f.Name(), f.Dirs()
+}
+
+// Convert the item to URL representation, ignore empty entries, adding prefix tokens as provided
+func (f *PluginFileItem) toURL(previousTokens ...string) string {
+	return toURL(toURL(previousTokens...), toURL(*f...))
+}
```

Comment:
Let's use Golang's [net/url](https://pkg.go.dev/net/url) package to get verification for the URL
```suggestion
func (f *PluginFileItem) toURL(rawUrl string) string {
	url, err := url.Parse(rawUrl)
	if err != nil {
		return "", err
	}
	url.Path = path.Join(url.Path, path.Join(*f...))
	return url.String(), nil
}
```

Reply:
didn't know this package, I will refactor and use this